src/com/boxboat/jenkins/library/docker/ArtifactoryClean.groovy

```diff
@@ -0,0 +1,120 @@
+package com.boxboat.jenkins.library.docker
+
+import com.boxboat.jenkins.library.config.Config
+
+class ArtifactoryClean implements Serializable {
+
+    boolean dryRun
+    int retentionDays
+    int removalCount = 0
+    List<ArtifactoryCleanRegistry> artifactoryCleanRegistries
+    String cleanSummary
+
+    Map<String, List<ArtifactoryImageManifest>> readRepositories(ArtifactoryCleanRegistry cleanRegistry) {
+        def registry = Config.global.getRegistry(cleanRegistry.registryKey)
+        def requestURI = registry.getRegistryUrl() + "/artifactory/api/search/aql"
+        def pathMatch = cleanRegistry.path == null ? "*" : cleanRegistry.path
```

Comment:
nitpick - use the Elvis Operator - shorter and also matches empty string

```
def pathMatch = cleanRegistry.path ?: "*"
```

Reply:
think we should rename `cleanRegistry.path` to `cleanRegistry.pathMatch`?

Reply:
that's probably a good idea